apps/extension/src/router/Message.ts

```diff
@@ -0,0 +1,35 @@
+import { Env, MessageSender } from "./types";
+
+/**
+ * This messaging system is influenced by cosmos-sdk.
+ * The messages are processed in the following order:
+ * "deserialize message -> approve external -> validate basic -> handler by routing".
+ * This deserializing system has weak polymorphism feature.
+ * Message would be converted to object according to their class and registered type.
+ * But, nested class is not supported. Non primitivie types or array that includes non primitive types in message's fields
+ * can't be decoded to their type properly. In this case, user should set thier prototype manually.
```

Comment:
```suggestion
 * can't be decoded to their type properly. In this case, user should set their prototype manually.
```

Reply:
Thanks! I actually rewrote the whole thing to make better sense. Some of the comments have been inherited, so it's good that we update them to our standards and according to our specifications!